### PR TITLE
[kube-prometheus-stack] Ability for custom dnsConfig

### PR DESCRIPTION
**Pull request by stafot**
#### What this PR does / why we need it:
Split of https://github.com/prometheus-community/helm-charts/pull/554

#### Which issue this PR fixes
Part of https://github.com/prometheus-community/helm-charts/issues/634
Adding possibility to set a custom dnsConfig for the pods. Pod's DNS Config field enable users more control on the DNS settings for the pods.
#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml [hunk 1]
```diff
@@ -63,6 +63,10 @@ spec:
 {{- if .Values.alertmanager.alertmanagerSpec.routePrefix }}
   routePrefix: "{{ .Values.alertmanager.alertmanagerSpec.routePrefix }}"
 {{- end }}
+{{- if .Values.alertmanager.alertmanagerSpec.dnsConfig }}
+dnsConfig:
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml:67]: indent looks wrong

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by bismarck — CHANGES_REQUESTED** [review]
`dnsConfig` does not exist in either [AlertmanagerSpec](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#alertmanagerspec) or [PrometheusSpec](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#prometheusspec).

File: charts/kube-prometheus-stack/values.yaml [hunk 2]
```diff
@@ -2145,6 +2154,7 @@ prometheus:
       runAsUser: 1000
       fsGroup: 2000
 
+      #   - name: edns0
```
Comment (Xtigyro) [inline charts/kube-prometheus-stack/values.yaml:2157]: @stafot Thanks for the contribution!

Should be a final question: Why is this added here? Or it's a leftover?

**Review by Xtigyro — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/values.yaml (hunk 2 above)
Comment (stafot) [reply charts/kube-prometheus-stack/values.yaml:2157]: @Xtigyro Yes it is leftover. Sorry for this. I will fix as soon as possible

**Review by stafot — COMMENTED** [review]
Thanks

**Review by Xtigyro — APPROVED** [review]

**Comment by Xtigyro** [comment]
@stafot You're welcome - and thanks once more! 🎖️ 

@monotek @bismarck Guys, are you OK with the PR now?

**Review by monotek — DISMISSED** [review]

**Review by bismarck — APPROVED** [review]
lgtm
